File: samples/dotnet-isolated/RedisOutputBinding/StreamTriggerDeleteEntry.cs
using Microsoft.Extensions.Logging;
using Newtonsoft.Json;
using StackExchange.Redis;
using System.Linq;

namespace Microsoft.Azure.Functions.Worker.Extensions.Redis.Samples.RedisOutputBinding
{
    internal class StreamTriggerDeleteEntry
    {
        [Function(nameof(StreamTriggerDeleteEntry))]
        [RedisOutput(Common.connectionStringSetting, "XDEL")]
        public static string Run(
            [RedisStreamTrigger(Common.connectionStringSetting, "streamTest2")] StreamEntry entry,
            ILogger logger)
        {
            logger.LogInformation($"Stream entry from key 'streamTest2' with Id '{entry.Id}' and values '" +
                JsonConvert.SerializeObject(entry.Values.ToDictionary(x => x.Name.ToString(), x => x.Value.ToString())));
            return "streamTest2 " + entry.Id.ToString();
        }
    }
}

File: samples/java/src/main/java/com/function/RedisOutputBinding/StreamTriggerDeleteEntry.java
package com.function.RedisOutputBinding;

import com.function.RedisStreamEntry;
import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.redis.annotation.*;

public class StreamTriggerDeleteEntry {
    @FunctionName("StreamTriggerDeleteEntry")
    public void run(
            @RedisStreamTrigger(
                name = "entry",
                connection = "redisConnectionString",
                key = "streamTest2")
                RedisStreamEntry entry,
            @RedisOutput(
                name = "value",
                connection = "redisConnectionString",
                command = "DEL")
                OutputBinding<String> value,
            final ExecutionContext context) {
                
                context.getLogger().info("Stream entry from key 'streamTest2' with Id '" + entry.Id + "'");
                value.setValue("streamTest2 " + entry.Id);
    }
}
